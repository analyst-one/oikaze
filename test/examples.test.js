/**
 * @jest-environment jest-environment-node-single-context
 */

const sass = require('sass');
const loadPaths = ['./'];

it('renders css for single theme', () => {
  const { css } = sass.compile('examples/custom/single.scss', { loadPaths });
  expect(css).toMatchSnapshot();
});

it('renders css for theme overrides', () => {
  const { css } = sass.compile('examples/custom/multi.scss', { loadPaths });
  expect(css).toMatchSnapshot();
});

it('renders css with prefixes', () => {
  const { css } = sass.compile('examples/custom/prefix.scss', { loadPaths });
  expect(css).toMatchSnapshot();
});

it('renders css with references', () => {
  const { css } = sass.compile('examples/three-tier/style.scss', { loadPaths });
  expect(css).toMatchSnapshot();
});

it('can generated classes', () => {
  const { css } = sass.compile('examples/custom/utilities.scss', { loadPaths });
  expect(css).toMatchSnapshot();
});

