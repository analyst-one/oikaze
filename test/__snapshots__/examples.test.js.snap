// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can generated classes 1`] = `
".color-primary-200 {
  color: var(--color-primary-200, #aed051);
}

.bg-primary-200 {
  background-color: var(--color-primary-200, #aed051);
}

.color-primary-300 {
  color: var(--color-primary-300, #c3dd7b);
}

.bg-primary-300 {
  background-color: var(--color-primary-300, #c3dd7b);
}

.color-primary-400 {
  color: var(--color-primary-400, #d7e9a6);
}

.bg-primary-400 {
  background-color: var(--color-primary-400, #d7e9a6);
}

.color-primary-600 {
  color: var(--color-primary-600, #75912b);
}

.bg-primary-600 {
  background-color: var(--color-primary-600, #75912b);
}

.color-primary-500 {
  color: var(--color-primary-500, #93b733);
}

.bg-primary-500 {
  background-color: var(--color-primary-500, #93b733);
}

.color-primary-700 {
  color: var(--color-primary-700, #576b21);
}

.bg-primary-700 {
  background-color: var(--color-primary-700, #576b21);
}

.color-primary-800 {
  color: var(--color-primary-800, #3a4717);
}

.bg-primary-800 {
  background-color: var(--color-primary-800, #3a4717);
}

.color-neutral-600 {
  color: var(--color-neutral-600, white);
}

.bg-neutral-600 {
  background-color: var(--color-neutral-600, white);
}

.color-neutral-500 {
  color: var(--color-neutral-500, rgb(255, 255, 255));
}

.bg-neutral-500 {
  background-color: var(--color-neutral-500, rgb(255, 255, 255));
}

.color-neutral-400 {
  color: var(--color-neutral-400, rgb(255, 255, 255));
}

.bg-neutral-400 {
  background-color: var(--color-neutral-400, rgb(255, 255, 255));
}

.color-neutral-300 {
  color: var(--color-neutral-300, #fefefe);
}

.bg-neutral-300 {
  background-color: var(--color-neutral-300, #fefefe);
}

.color-neutral-200 {
  color: var(--color-neutral-200, #fefefe);
}

.bg-neutral-200 {
  background-color: var(--color-neutral-200, #fefefe);
}

.color-neutral-100 {
  color: var(--color-neutral-100, #fefefe);
}

.bg-neutral-100 {
  background-color: var(--color-neutral-100, #fefefe);
}

.color-neutral-800 {
  color: var(--color-neutral-800, #7f7f7f);
}

.bg-neutral-800 {
  background-color: var(--color-neutral-800, #7f7f7f);
}

.color-neutral-900 {
  color: var(--color-neutral-900, black);
}

.bg-neutral-900 {
  background-color: var(--color-neutral-900, black);
}

.color-primary {
  color: var(--color-primary, #93b733);
}

.bg-primary {
  background-color: var(--color-primary, #93b733);
}

.color-neutral {
  color: var(--color-neutral, #fefefe);
}

.bg-neutral {
  background-color: var(--color-neutral, #fefefe);
}

.font-none {
  font-size: 0rem;
}

.font-xx-small {
  font-size: 0.125rem;
}

.font-x-small {
  font-size: 0.25rem;
}

.font-small {
  font-size: 0.5rem;
}

.font-regular {
  font-size: 1rem;
}

.font-large {
  font-size: 2rem;
}

.font-x-large {
  font-size: 4rem;
}

.font-xx-large {
  font-size: 8rem;
}

.opacity-20 {
  opacity: 20%;
}

.opacity-50 {
  opacity: 50%;
}

.opacity-80 {
  opacity: 80%;
}"
`;

exports[`renders css for single theme 1`] = `
":root {
  --color-primary-200: #aed051;
  --color-primary-300: #c3dd7b;
  --color-primary-400: #d7e9a6;
  --color-primary-600: #75912b;
  --color-primary-500: #93b733;
  --color-primary-700: #576b21;
  --color-primary-800: #3a4717;
  --color-neutral-600: white;
  --color-neutral-500: rgb(255, 255, 255);
  --color-neutral-400: rgb(255, 255, 255);
  --color-neutral-300: #fefefe;
  --color-neutral-200: #fefefe;
  --color-neutral-100: #fefefe;
  --color-neutral-800: #7f7f7f;
  --color-neutral-900: black;
  --color-primary: #93b733;
  --color-neutral: #fefefe;
  --size-none: 0;
  --size-none--em: 0;
  --size-xx-small: 2px;
  --size-xx-small--em: 0.125;
  --size-x-small: 4px;
  --size-x-small--em: 0.25;
  --size-small: 0.5;
  --size-small--em: 0.5;
  --size-regular: 16px;
  --size-regular--em: 1;
  --size-large: 200%;
  --size-large--em: 2;
  --size-x-large: 64px;
  --size-x-large--em: 4;
  --size-xx-large: 128px;
  --size-xx-large--em: 8;
  --opacity-20: 0.2;
  --opacity-20--em: 0.2;
  --opacity-50: 50%;
  --opacity-50--em: 0.5;
  --opacity-80: 80%;
  --opacity-80--em: 0.8;
}

.test {
  color: #93b733;
  color: var(--color-primary, #93b733);
  color: color-mix(in srgb, var(--color-primary, #93b733) 20%, transparent);
  color: rgba(147, 183, 51, 0.2);
  font-size: var(--size-regular, 16px);
  font-size: 16px;
  font-size: calc(var(--size-small--em, 0.5) * 1rem);
  font-size: 0.5rem;
}"
`;

exports[`renders css for theme overrides 1`] = `
":root {
  --color-primary-200: #aed051;
  --color-primary-300: #c3dd7b;
  --color-primary-400: #d7e9a6;
  --color-primary-600: #75912b;
  --color-primary-500: #93b733;
  --color-primary-700: #576b21;
  --color-primary-800: #3a4717;
  --color-neutral-600: white;
  --color-neutral-500: rgb(255, 255, 255);
  --color-neutral-400: rgb(255, 255, 255);
  --color-neutral-300: #fefefe;
  --color-neutral-200: #fefefe;
  --color-neutral-100: #fefefe;
  --color-neutral-800: #7f7f7f;
  --color-neutral-900: black;
  --color-primary: #93b733;
  --color-neutral: #fefefe;
  --size-none: 0;
  --size-none--em: 0;
  --size-xx-small: 2px;
  --size-xx-small--em: 0.125;
  --size-x-small: 4px;
  --size-x-small--em: 0.25;
  --size-small: 0.5;
  --size-small--em: 0.5;
  --size-regular: 16px;
  --size-regular--em: 1;
  --size-large: 200%;
  --size-large--em: 2;
  --size-x-large: 64px;
  --size-x-large--em: 4;
  --size-xx-large: 128px;
  --size-xx-large--em: 8;
  --opacity-20: 0.2;
  --opacity-20--em: 0.2;
  --opacity-50: 50%;
  --opacity-50--em: 0.5;
  --opacity-80: 80%;
  --opacity-80--em: 0.8;
}

body.alt {
  --color-primary: #ff0000;
  --size-root: 14px;
  --size-root--em: 1;
  --size-small: 7px;
  --size-small--em: 0.5;
  --size-regular: 14px;
  --size-regular--em: 1;
}

.test {
  color: #ff0000;
  color: var(--color-primary, #ff0000);
  color: color-mix(in srgb, var(--color-primary, #ff0000) 20%, transparent);
  color: rgba(255, 0, 0, 0.2);
  font-size: calc(var(--size-small--em, 0.5) * 1rem);
  font-size: 0.5rem;
}"
`;

exports[`renders css with prefixes 1`] = `
":root {
  --ok--color-primary-200: #aed051;
  --ok--color-primary-300: #c3dd7b;
  --ok--color-primary-400: #d7e9a6;
  --ok--color-primary-600: #75912b;
  --ok--color-primary-500: #93b733;
  --ok--color-primary-700: #576b21;
  --ok--color-primary-800: #3a4717;
  --ok--color-neutral-600: white;
  --ok--color-neutral-500: rgb(255, 255, 255);
  --ok--color-neutral-400: rgb(255, 255, 255);
  --ok--color-neutral-300: #fefefe;
  --ok--color-neutral-200: #fefefe;
  --ok--color-neutral-100: #fefefe;
  --ok--color-neutral-800: #7f7f7f;
  --ok--color-neutral-900: black;
  --ok--color-primary: #93b733;
  --ok--color-neutral: #fefefe;
  --ok--size-none: 0;
  --ok--size-none--em: 0;
  --ok--size-xx-small: 2px;
  --ok--size-xx-small--em: 0.125;
  --ok--size-x-small: 4px;
  --ok--size-x-small--em: 0.25;
  --ok--size-small: 0.5;
  --ok--size-small--em: 0.5;
  --ok--size-regular: 16px;
  --ok--size-regular--em: 1;
  --ok--size-large: 200%;
  --ok--size-large--em: 2;
  --ok--size-x-large: 64px;
  --ok--size-x-large--em: 4;
  --ok--size-xx-large: 128px;
  --ok--size-xx-large--em: 8;
  --ok--opacity-20: 0.2;
  --ok--opacity-20--em: 0.2;
  --ok--opacity-50: 50%;
  --ok--opacity-50--em: 0.5;
  --ok--opacity-80: 80%;
  --ok--opacity-80--em: 0.8;
}

.test {
  color: #93b733;
  color: var(--ok--color-primary, #93b733);
  color: color-mix(in srgb, var(--ok--color-primary, #93b733) 20%, transparent);
  color: rgba(147, 183, 51, 0.2);
  font-size: calc(var(--ok--size-small--em, 0.5) * 1rem);
  font-size: 0.5rem;
}"
`;

exports[`renders css with references 1`] = `
"body {
  --color-primary: #2a4ff0;
  --color-secondary: #ff593a;
  --color-foreground: #000000;
  --color-background: #ffffff;
  color: var(--color-foreground, #000000);
  background-color: var(--color-background, #ffffff);
}
body.invert {
  --color-primary: #4557b6;
  --color-secondary: #a1170a;
  --color-foreground: #ffffff;
  --color-background: #000000;
}

@media (prefers-color-scheme: dark) {
  body {
    --color-primary: #4557b6;
    --color-secondary: #a1170a;
    --color-foreground: #ffffff;
    --color-background: #000000;
  }
  body.invert {
    --color-primary: #2a4ff0;
    --color-secondary: #ff593a;
    --color-foreground: #000000;
    --color-background: #ffffff;
  }
}
.btn {
  --button-primary-background: var(--color-primary, #2a4ff0);
  --button-primary-text: var(--color-foreground, #000000);
  --button-secondary-background: var(--color-secondary, #ff593a);
  --button-secondary-text: var(--color-foreground, #000000);
}
.btn--primary {
  background-color: var(--button-primary-background, #2a4ff0);
  color: var(--button-primary-text, #000000);
}
.btn--secondary {
  background-color: var(--button-secondary-background, #ff593a);
  color: var(--button-secondary-text, #000000);
}"
`;
